shell/models/__tests__/management.cattle.io.node.ts

```diff
@@ -0,0 +1,82 @@
+import MgmtNode from '@shell/models/management.cattle.io.node';
+
+describe('class MgmtNode', () => {
+  const t = jest.fn(() => 'baz');
+  const ctx = { rootGetters: { 'i18n/t': t } };
+
+  const resetMocks = () => {
+    // Clear all mock function calls:
+    jest.clearAllMocks();
+  };
+
+  it('should not return addresses if they are not present in the resource status, the internalNodeStatus, or the rkeNode key in status', () => {
+    const mgmtNode = new MgmtNode({ status: {} }, ctx);
+
+    expect(mgmtNode.addresses).toStrictEqual([]);
+    resetMocks();
+  });
+  it('should return addresses if they are present directly on the resource status', () => {
+    const mgmtNode = new MgmtNode({ status: { addresses: ['foo'] } }, ctx);
+
+    expect(mgmtNode.addresses).toStrictEqual(['foo']);
+    resetMocks();
+  });
+  it('should return addresses if they are not present directly on the resource status but are on "status.internalNodeStatus"', () => {
+    const mgmtNode = new MgmtNode({ status: { internalNodeStatus: { addresses: ['bar'] } } }, ctx);
+
+    expect(mgmtNode.addresses).toStrictEqual(['bar']);
+    resetMocks();
+  });
+  it('should return an internalIp if addresses includes an object with an appropriate type and address', () => {
+    const mgmtNode = new MgmtNode({ status: { addresses: [{ type: 'InternalIP', address: 'foo' }] } }, ctx);
+
+    expect(mgmtNode.internalIp).toStrictEqual('foo');
+    resetMocks();
+  });
+  it('should return an externalIp if addresses includes an object with an appropriate type and address', () => {
+    const mgmtNode = new MgmtNode({ status: { addresses: [{ type: 'ExternalIP', address: 'bar' }] } }, ctx);
+
+    expect(mgmtNode.externalIp).toStrictEqual('bar');
+    resetMocks();
+  });
+  it('should return an internalIp if internalNodeStatus.addresses includes an object with an appropriate type and address', () => {
+    const mgmtNode = new MgmtNode({ status: { internalNodeStatus: { addresses: [{ type: 'InternalIP', address: 'foo' }] } } }, ctx);
+
+    expect(mgmtNode.internalIp).toStrictEqual('foo');
+    resetMocks();
+  });
+  it('should return an externalIp if internalNodeStatus.addresses includes an object with an appropriate type and address', () => {
```

Comment:
This test can be parametrized.

Reply:
It could be and if there were enough permutations to it, I would parameterize it and run several variables across eachother but in this case, there's not too many use cases, only a single variable, and most of the tests are incredibly simple so the added overhead of parameterizing the test and looping through test conditions seems like it would do more harm than good from a readability perspective and towards expanding this test suite for the remaining model.